### PR TITLE
Feature: add new Studio CommandLine

### PR DESCRIPTION
**Pull request by markexin**
### <samp>🤖 Generated by Copilot at aace4eb</samp>

Added a feature to disable browser opening for the `studio` and `test` commands in the `maestro-cli` module. This gives the user more control over the workflow and avoids unwanted browser windows.

Hello author, I have encountered a scene that I hope to integrate into vscode in the form of plug-ins, but when I start Maestro Studio, the browser will start and load the page uncontrollably, which I personally think may not be very friendly, I hope this place can give me free control.

### CODE REVIEW

**Comment by markexin** [comment]
> @markexin : Does this solve this problem? #1174
yes~~~great!!!

**Comment by markexin** [comment]
Hello author, after hard work, my code passed the test. Please kindly help to Review if there are any defects

**Comment by axelniklasson** [comment]
Hi there @markexin, thanks for taking the time to create a pull request. Could you share more about this use case? In what case would a browser be opened when you run `maestro test`? Are you writing Flows using the [WebDriver](https://maestro.mobile.dev/advanced/experimental/maestro-webdriver)?

**Comment by markexin** [comment]
> Hi there @markexin, thanks for taking the time to create a pull request. Could you share more about this use case? In what case would a browser be opened when you run `maestro test`? Are you writing Flows using the [WebDriver](https://maestro.mobile.dev/advanced/experimental/maestro-webdriver)?

Thank you for the author's reminder. For example:

When I execute:

`maestro studio` or `maestro studio --nowindow`  will automatically open the browser at this time.

Only when I clearly execute:

`maestro studio --nowindow=false` will not automatically open the browser.

**Comment by axelniklasson** [comment]
> > Hi there @markexin, thanks for taking the time to create a pull request. Could you share more about this use case? In what case would a browser be opened when you run `maestro test`? Are you writing Flows using the [WebDriver](https://maestro.mobile.dev/advanced/experimental/maestro-webdriver)?
> 
> Thank you for the author's reminder. For example:
> 
> When I execute:
> 
> `maestro studio` or `maestro studio --window` will automatically open the browser at this time.
> 
> Only when I clearly execute:
> 
> `maestro studio --window=false` will not automatically open the browser.

Could you elaborate a bit more as to why you would like to open Studio without opening a browser window at the same time?

File: maestro-cli/src/main/java/maestro/cli/runner/TestSuiteInteractor.kt [hunk 1]
```diff
@@ -26,6 +26,8 @@ class TestSuiteInteractor(
     private val maestro: Maestro,
```
[the remark this answers is not in the document]
Comment (markexin) [reply maestro-cli/src/main/java/maestro/cli/runner/TestSuiteInteractor.kt]: Done

File: maestro-cli/src/main/java/maestro/cli/command/StudioCommand.kt [hunk 2]
```diff
@@ -35,6 +35,12 @@ class StudioCommand : Callable<Int> {
     )
     private var debugOutput: String? = null
 
+    @CommandLine.Option(
+        names = ["-nw"],
+        description = ["Configures automatically start browser, default is false"]
```
[the remark this answers is not in the document]
Comment (markexin) [reply maestro-cli/src/main/java/maestro/cli/command/StudioCommand.kt]: Done

File: maestro-cli/src/main/java/maestro/cli/command/TestCommand.kt [hunk 3]
```diff
@@ -87,6 +87,12 @@ class TestCommand : Callable<Int> {
     )
```
[the remark this answers is not in the document]
Comment (axelniklasson) [reply maestro-cli/src/main/java/maestro/cli/command/TestCommand.kt]: Please remove the file from the PR altogether as there is just whitespace change

File: maestro-cli/src/main/java/maestro/cli/runner/TestSuiteInteractor.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (axelniklasson) [reply maestro-cli/src/main/java/maestro/cli/runner/TestSuiteInteractor.kt]: Please remove the file from the PR altogether as there is just whitespace change

**Review by axelniklasson — APPROVED** [review]
Thanks for the contribution @markexin! Let us know if you end up building some IDE integration :)

**Comment by markexin** [comment]
> Thanks for the contribution @markexin! Let us know if you end up building some IDE integration :)


Thank you for your patience and guidance, I hope the project will get better and better

**Pull request merged**